weasyl/image.py

```diff
@@ -50,74 +14,24 @@ def image_setting(im):
         return 'G'
 
 
-def image_file_type(im):
-    ret = image_extension(im)
-    if ret is not None:
-        ret = ret.lstrip('.')
-    return ret
-
-
-def check_crop(dim, x1, y1, x2, y2):
+def check_crop(image_size, crop_bounds):
     """
-    Return True if the specified crop coordinates are valid, else False.
+    Return True if the specified crop rectangle is valid, else False.
     """
-    return (
-        x1 >= 0 and y1 >= 0 and x2 >= 0 and y2 >= 0 and x1 <= dim[0] and
-        y1 <= dim[1] and x2 <= dim[0] and y2 <= dim[1] and x2 > x1 and y2 > y1)
+    image_bounds = image_size.at(geometry.origin)
+    return bool(crop_bounds) and crop_bounds in image_bounds
 
 
-def check_type(filename, secure=True):
+def check_type(filename):
     """
     Return True if the filename corresponds to an image file, else False.
     """
-    if secure:
-        try:
-            im = Image.read(filename)
-        except SanperaError:
-            return False
-        else:
-            return im.original_format in ['JPEG', 'PNG', 'GIF']
+    try:
+        im = Image.read(filename)
+    except SanperaError:
+        return False
     else:
-        return filename and filename[-4:] in [".jpg", ".png", ".gif"]
-
-
-def _resize(im, width, height):
-    """
-    Resizes the image to fit within the specified height and width; aspect ratio
-    is preserved. Images always preserve animation and might even result in a
-    better-optimized animated gif.
-    """
-    # resize only if we need to; return None if we don't
-    if im.size.width > width or im.size.height > height:
-        im = im.resized(im.size.fit_inside((width, height)))
-        return im
-
-
-def resize(filename, width, height, destination=None, animate=False):
-    in_place = False
-    if not destination:
-        destination = filename + '.new'
-        in_place = True
-
-    im = read(filename)
-    if not image_extension(im):
-        raise WeasylError("FileType")
-
-    files.ensure_file_directory(filename)
-    im = correct_image_and_call(_resize, im, width, height)
-    if im is not None:
-        im.write(destination)
-        if in_place:
-            os.rename(destination, filename)
-
-    # if there's no need to resize, in-place resize is a no-op. otherwise copy
-    # the source to the destination.
-    elif not in_place:
-        files.copy(filename, destination)
-
-
-def resize_image(im, width, height):
-    return correct_image_and_call(_resize, im, width, height) or im
+        return im.original_format in ['JPEG', 'PNG', 'GIF']
```

Comment:
I’d like to err on the side of maintaining existing behaviour for this pull request, and just remove all the character special casing in a future one.